x/incentive/keeper/payout.go

```diff
@@ -47,7 +47,56 @@ func (k Keeper) ClaimReward(ctx sdk.Context, addr sdk.AccAddress, multiplierName
 		return err
 	}
 
-	k.ZeroClaim(ctx, claim)
+	k.ZeroUSDXMintingClaim(ctx, claim)
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeClaim,
+			sdk.NewAttribute(types.AttributeKeyClaimedBy, addr.String()),
+			sdk.NewAttribute(types.AttributeKeyClaimAmount, claim.Reward.String()),
+		),
+	)
+	return nil
+}
+
+// ClaimHardReward sends the reward amount to the input address and zero's out the claim in the store
+func (k Keeper) ClaimHardReward(ctx sdk.Context, addr sdk.AccAddress, multiplierName types.MultiplierName) error {
+	_, found := k.GetHardLiquidityProviderClaim(ctx, addr)
+	if !found {
+		return sdkerrors.Wrapf(types.ErrClaimNotFound, "address: %s", addr)
+	}
+
+	multiplier, found := k.GetMultiplier(ctx, multiplierName)
+	if !found {
+		return sdkerrors.Wrapf(types.ErrInvalidMultiplier, string(multiplierName))
+	}
+
+	claimEnd := k.GetClaimEnd(ctx)
+
+	if ctx.BlockTime().After(claimEnd) {
+		return sdkerrors.Wrapf(types.ErrClaimExpired, "block time %s > claim end time %s", ctx.BlockTime(), claimEnd)
+	}
+
+	k.SynchronizeHardLiquidityProviderClaim(ctx, addr)
+
+	claim, found := k.GetHardLiquidityProviderClaim(ctx, addr)
+	if !found {
+		return sdkerrors.Wrapf(types.ErrClaimNotFound, "address: %s", addr)
+	}
+
+	rewardAmount := claim.Reward.Amount.ToDec().Mul(multiplier.Factor).RoundInt()
+	if rewardAmount.IsZero() {
+		return types.ErrZeroClaim
+	}
+	rewardCoin := sdk.NewCoin(claim.Reward.Denom, rewardAmount)
+	length := ctx.BlockTime().AddDate(0, int(multiplier.MonthsLockup), 0).Unix() - ctx.BlockTime().Unix()
+
+	err := k.SendTimeLockedCoinsToAccount(ctx, types.IncentiveMacc, addr, sdk.NewCoins(rewardCoin), length)
+	if err != nil {
+		return err
+	}
+
+	k.ZeroHardLiquidityProviderClaim(ctx, claim)
```

Comment:
ditto add claim type to event

Reply:
https://github.com/Kava-Labs/kava/pull/780/commits/ac6d43f4968d631eb510ab7cd7fee7ae36bdfc7f